src/linear_algebra/dual.lean

```diff
@@ -43,7 +37,7 @@ variables {K : Type u} {V : Type v} {ι : Type w}
 variables [discrete_field K] [add_comm_group V] [vector_space K V]
 open vector_space module module.dual submodule linear_map cardinal function
 
-instance dual.vector_space : vector_space K (dual K V) := {..dual.module K V}
+instance dual.vector_space : vector_space K (dual K V) := { ..module.dual.inst K V }
```

Comment:
It makes an attempt to use something close to the standard. There are a few issues. Here, the attempt fails because the type has two arguments, so it falls back to the default. But even when it succeeds (which it will for e.g. `ring` instances), it appends something to the end. This is because we can't protect the instance names when we add them this way, and sometimes the default names will conflict with other identifiers when namespaces are open. 

This confused me for a while when I was making these changes:
```lean
@[derive category]
def over (X : T) := comma.{v₃ 0 v₃} (𝟭 T) (functor.of.obj X)
```
was trying to add an instance named `category_theory.over.category`. But since the namespace `category_theory.over` was open, all following applications of `category` resolved to this instance, which was a mess.

I don't think we can detect when this is likely to happen. So it's necessary to put some nonsense in the instance names to avoid it.

Reply:
It's unfortunate we cannot protect declarations via meta-programming. This indeed seems to be the best option.